### PR TITLE
add prometheus federate->statsd forwarder

### PR DESCRIPTION
**Pull request by jim-minter**
@mjudeikis ptal

### CODE REVIEW

File: cmd/metricsbridge/metricsbridge.go [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+		for _, m := range family.Metric {
+			gauge := &statsd.Gauge{
+				Namespace: c.Namespace,
+				Metric:    *family.Name,
+				Dims: map[string]string{
+					"Region":       c.Region,
+					"UnderlayName": "",
+				},
+				Value: *m.Untyped.Value,
```
Comment (mjudeikis) [inline cmd/metricsbridge/metricsbridge.go]:
```
GOOD:
........j-.{"Namespace":"Prometheus","Metric":"kube_node_info","Dims":{"Region":"eastus","UnderlayName":"","cluster":"https://mjudeikis-metrics.eastus.cloudapp.azure.com","container_runtime_version":"docker://1.13.1","endpoint":"https-main","instance":"10.131.0.14:8443","job":"kube
-state-metrics","kernel_version":"3.10.0-957.el7.x86_64","kubelet_version":"v1.11.0+d4cacc0","kubeproxy_version":"v1.11.0+d4cacc0","namespace":"openshift-monitoring","node":"master-000000","os_image":"Red Hat Enterprise Linux Server 7.6 (Maipo)","pod":"kube-state-metrics-64cc8b86b5
-xt948","prometheus":"openshift-monitoring/k8s","prometheus_replica":"prometheus-k8s-0","provider_id":"azure:///subscriptions/225e02bc-43d0-43d1-a01a-17e584a4ef69/resourceGroups/mjudeikis-metrics/providers/Microsoft.Compute/virtualMachineScaleSets/ss-master/virtualMachines/0","serv
ice":"kube-state-metrics"}}:1|g

CURRENT:
........[".{"__name__":"kube_node_status_condition","cluster":"https://mjudeikis-metrics.eastus.cloudapp.azure.com","condition":"Ready","endpoint":"https-main","instance":"10.131.0.14:8443","job":"kube-state-metrics","namespace":"openshift-monitoring","node":"master-000002","pod":"
kube-state-metrics-64cc8b86b5-xt948","prometheus":"openshift-monitoring/k8s","prometheus_replica":"prometheus-k8s-1","service":"kube-state-metrics","status":"unknown"}:0.000000|g                                                                                                        
```

After speaking with @sozercan and some TCPDUMP'int - converted pod is converting all `float64` to `int64`. We could try not to convert and see how MDM handles those. From a metrics point of I would like to keep a float, as it gives more freedom in the calculation space.

Comment (mjudeikis) [reply cmd/metricsbridge/metricsbridge.go]: Ignore. Seen the conversion now.

**Comment by mjudeikis** [comment]
Happy with the change. Will need to go line by line all this as I need to understand this better :) Will update release tests too for this.

File: pkg/util/statsd/statsd.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	buf := &bytes.Buffer{}
+
+	e := json.NewEncoder(buf)
+	err := e.Encode(g)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: would expect not to be converting to an int here, let's see...
+	_, err = fmt.Fprintf(buf, ":%d|g\n", int(g.Value))
```
[the remark this answers is not in the document]
Comment (jim-minter) [reply pkg/util/statsd/statsd.go]: should be fixed

**Comment by jim-minter** [comment]
@mjudeikis please lgtm if happy

**Pull request merged**